src/MusicTheory/NoteOrTuplet.elm

```diff
@@ -0,0 +1,438 @@
+module MusicTheory.NoteOrTuplet exposing
+    ( NoteOrTuplet
+    , duplet
+    , eighth
+    , half
+    , mapAll
+    , mapFifth
+    , mapFirst
+    , mapFourth
+    , mapSecond
+    , mapThird
+    , note
+    , oneHundredTwentyEighth
+    , quadruplet
+    , quintuplet
+    , setDivision
+    , setDotted
+    , setDoubleDotted
+    , setPitches
+    , setRest
+    , setSeparate
+    , setTied
+    , setTripleDotted
+    , sixteenth
+    , sixtyFourth
+    , thirtySecond
+    , triplet
+    , whole
+    )
+
+import MusicTheory.Pitch exposing (Pitch)
+
+
+type NoteOrTuplet
```

Comment:
I don't like this name, but it's hard to name something that's this general. this type can represent:
- A note
- A rest
- A chord / multiple simultaneous notes of the same division
- Any combination of these grouped in a tuplet of 2–5

Maybe this is more properly a `NoteValue`? i.e. https://en.wikipedia.org/wiki/Note_value

Reply:
Also, I was working under the assumption that a `Line` or whatever we call a bunch of "note events" on a staff would be something like a `List NoteOrTuplet`. But maybe `Note`s and `Tuplet`s should be their own types, and they get unified into a `Line` with another wrapper type. Open to your ideas here.

Reply:
Yes, that's really hard.

I'm in favour of giving it a generic parameter, so we can put anything in there.

Because maybe we don't want to represent a chord as a list of pitches but rather as something else...

What comes to mind is `Token` but thats too generic.

`Tuplet` is in a way ok, because a single note is like a one-tuplet. On the other hand this is not nice because single ones are probably default and then it's weird when everything is called tuplet.

What a bout `Term`?

Or `RhythmElement`?